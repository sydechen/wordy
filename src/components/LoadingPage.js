import React from 'react';

const LoadingPage = () => (
    <div className='loader'>
        <img className='loader__image' src='/img/loading.gif' />
    </div>
);

export default LoadingPage;