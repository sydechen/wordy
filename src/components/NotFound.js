import React from 'react';
import { Link } from 'react-router-dom';


const NotFound = () => (
    <div>
        404..you wanna go <Link to='/'>home?</Link>
    </div>
);

export default NotFound;