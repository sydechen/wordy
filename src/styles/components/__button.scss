.button {
    background: $green;
    border: none;
    color: white;
    display: inline-block;
    font-weight: 300;
    font-size: $f-large;
    line-height: 1;
    padding: $s-size $m-size;
    text-decoration: none;
}

.button--link {
    background: none;
}

.button--secondary {
    background: #888;
}