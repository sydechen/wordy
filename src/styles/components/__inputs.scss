.text-input {
    border: 1px solid #cacccd;
    height: 46px;
    padding: $s-size;
    font-size: $f-large;
    font-weight: 300;
}

.select {
    @extend .text-input;
}

.text-area {
    @extend .text-input;
    height: 10rem;
}