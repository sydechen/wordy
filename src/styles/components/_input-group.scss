.input-group {
    display: flex;
    flex-direction: row;
    margin-bottom: $l-size;

    @media (max-width: $desktop-bpoint) {
        flex-direction: column;
        margin-bottom: $m-size;
    }
}

.input-group__item {
    margin-right: $s-size;

    @media (max-width: $desktop-bpoint) {
        margin: 0 0 $s-size 0;
    }
}