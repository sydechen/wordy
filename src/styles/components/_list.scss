.list-header {
    background: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
}

.list-body {
    margin-bottom: $l-size;

    @media(max-width: $desktop-bpoint) {
        margin-bottom: $m-size;
    }
}

.list-item {
    align-items: center;
    display: flex;
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    padding: $m-size;
    flex-direction: row;
    justify-content: space-between;
    text-decoration: none;
    transition: background .3s ease;
    &:hover {
        background: $off-white;
    }

    @media (max-width: $desktop-bpoint) {
        align-items: normal;
        flex-direction: column;
        padding: $s-size;
    }
}

.list-item--message {
    align-items: center;
    justify-content: center;
    color: $grey;
    padding: $m-size;
    &:hover {
        background: none;
    }
}

.list-item__title {
    margin: 0;
    word-break: break-all;
}

.list-item__subtitle {
    color: $grey;
    font-size: $f-small;
}

.list-item__data {
    margin: 0;
    padding-left: $s-size;

    @media (max-width: $desktop-bpoint) {
        margin-top: $s-size;
    }
}