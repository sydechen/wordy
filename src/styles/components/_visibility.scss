.show-mobile {
    @media (min-width: $desktop-bpoint + .01rem) {
        display:none;
    }
}

.show-desktop {
    @media (max-width: $desktop-bpoint) {
        display:none;
    }
}