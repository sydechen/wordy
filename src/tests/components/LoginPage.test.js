import {LoginPage} from '../../components/LoginPage';
import React from 'react';
import {shallow} from 'enzyme';

let startLogin, history, wrapper;

beforeEach(() => {
    startLogin = jest.fn();
    history = {push:jest.fn()};
    wrapper = shallow(<LoginPage startLogin = {startLogin} history = {history}/>);
})

test('render Login page', () => {
    expect(wrapper).toMatchSnapshot();
})

test('call startLogout on button click', () => {
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});