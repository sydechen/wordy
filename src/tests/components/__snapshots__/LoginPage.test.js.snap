// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      App Title
    </h1>
    <p>
      App tagline
    </p>
    <button
      className="button"
      onClick={[MockFunction]}
    >
      login with google
    </button>
  </div>
</div>
`;
